### ISSUE TITLE
`finish` emitted before data is copied

### ISSUE THREAD

**Issue by Sequoia** [brianc/node-pg-copy-streams#48]
I'm running my code in AWS lambda, wherein you call `done` when you're process is ready to exit.

```
        /** @type {NodeJS.WritableStream} */
        var pgStream = client.query(copyFrom(copyQuery));

        csvStream
          .pipe(pgStream)
          .on('error', function(e){
            logerr('pg stream error');
            done(e);
          })
          .on('finish', function(){
            console.timeEnd('query time');
            done(); //////////////// <---- Offending line
          });
```

This always exits **before the data is copied**. If I remove `done`, the process won't exit but the data **does copy**. Why is finish getting called before data is streamed? Should I be listening for "finish" elsewhere?

**Comment by Sequoia** [comment]
use `end` instead. My issue is resolved (sorta), but if there's a good place to add this to the docs lmk, it would be really useful!

**Comment by jeromew** [comment]
What version of pgCopy did you test ?

`finish` does not give you information on the end of the COPY, simply that all the incoming data has fully entered the copyFrom Stream. It may be still flushing.

`end` should tell you when the process is finished but we may need to delay it to a later ACK of the database. If you have an error you can reproduce we can try and find the problem.

can you try the version that is on this branch ?

https://github.com/jeromew/node-pg-copy-streams/tree/upstream-end

In this version the `end` version is delayed after CommandComplete is received.

You might also have success with pg-copy-streams@1.1.0 because a modification that occured in 1.1.1 led to `end` being called 2 times : 1 time on CopyDone and 1 time on CommandComplete.

You may also try in your code, with the current npm version to call done() but only on the second 'end' received.

A fix is being discussed on #54

**Comment by Sequoia** [comment]
thx! Using 1.1.0. It works for now so I'll close this ticket.
